packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1539,81 +1570,65 @@ export default abstract class SqlIntegration
     return `LEAST(${cap}, ${value})`;
   }
 
-  private addPrePostTimeFilter(
+  private addCaseWhenTimeFilter(
     col: string,
-    timePeriod: MetricAggregationType
+    ignoreConversionEnd: boolean,
+    cumulativeDate: boolean
   ): string {
-    const mcol = `m.timestamp`;
-    if (timePeriod === "pre") {
-      return `${this.ifElse(`${mcol} < d.preexposure_end`, `${col}`, `NULL`)}`;
-    }
-    if (timePeriod === "post") {
-      return `${this.ifElse(
-        `${mcol} >= d.conversion_start`,
-        `${col}`,
-        `NULL`
-      )}`;
-    }
-    return `${col}`;
+    return `${this.ifElse(
+      `
+        m.timestamp >= d.conversion_start
+        ${ignoreConversionEnd ? "" : `AND m.timestamp <= d.conversion_end`}
+        ${
+          cumulativeDate ? `AND ${this.dateTrunc("m.timestamp")} <= dr.day` : ""
+        }
+      `,
+      // The coalesce treats conversions that exist but have a NULL `value` as 0,
+      // reserving NULL for users with no matching metric rows in conversion window
+      `COALESCE(${col}, 0)`,
+      `NULL`
+    )}`;
   }
 
-  private getAggregateMetricColumn(
-    metric: MetricInterface,
-    metricTimeWindow: MetricAggregationType = "post"
-  ) {
+  private getAggregateMetricColumn(metric: MetricInterface) {
     // Binomial metrics don't have a value, so use hard-coded "1" as the value
     if (metric.type === "binomial") {
-      return `MAX(${this.addPrePostTimeFilter("1", metricTimeWindow)})`;
+      return `MAX(COALESCE(value, 0))`;
     }
 
     // SQL editor
     if (this.getMetricQueryFormat(metric) === "sql") {
       // Custom aggregation that's a hardcoded number (e.g. "1")
       if (metric.aggregation && Number(metric.aggregation)) {
-        return `MAX(${this.addPrePostTimeFilter(
-          metric.aggregation,
-          metricTimeWindow
-        )})`;
+        return `GREATEST(${metric.aggregation}, COALESCE(value, -999999))`;
```

Comment:
Hmm, yeah this seems wrong to me. I was handling the case where the CASE WHEN in __userMetricJoin returns a NULL, but really I should return 0 if the value is NULL since we now know that `value IS NULL` if and only if the user has no conversions in the metric source (because we now force NULL to be == 0 in the metric source).